### PR TITLE
add min pooling

### PR DESCRIPTION
**Pull request by ashione**
for special application, we add min pooling implementation

### CODE REVIEW

**Comment by seanbell** [comment]
I think you made the changes based on an old version of caffe, since the diff includes a lot of unrelated changes.  Right now this can't be merged.

Check out the newest version of the master branch, and apply just the relevant commit (https://github.com/BVLC/caffe/pull/3998/commits/51d38a4b34bddffb00871ac8d8fd3020b7309132), and then make a PR based on that.

File: Makefile [hunk 1]
```diff
@@ -393,7 +393,7 @@ else
 	endif
 endif
 INCLUDE_DIRS += $(BLAS_INCLUDE)
-LIBRARY_DIRS += $(BLAS_LIB)
+LIBRARY_DIRS += $(BLAS_LIB) /usr/lib/atlas-base
```
Comment (ashione) [inline Makefile]: this configuration is just for myself.

**Comment by ashione** [comment]
main branch can merge pooling_layer.hpp, pooling_layer.cpp and pooling_layer.cu these 3 files but other files.

**Comment by seanbell** [comment]
Sure, but this PR cannot be merged as-is.  Please rebase if you want it merged into caffe.

**Pull request closed without merging**